flask_monitoringdashboard/views/testmonitor.py

```diff
@@ -1,73 +1,125 @@
+                'median-overall': -1,
+                'last-tested': get_value(tested_times, endpoint, default=None)
+            })
 
-        return render_template('fmd_testmonitor/testmonitor.html', tests=get_tests(db_session), endpoints=endp_names,
-                               results=get_results(db_session), groups=grouped, colors=cols,
-                               res_current_version=get_res_current(db_session, config.version),
-                               boxplot=get_boxplot(None))
+        return render_template('fmd_testmonitor/testmonitor.html', result=result)
 
 
-def get_boxplot(test):
+def get_boxplot(endpoint=None, form=None):
+    """
+    Generates a box plot visualization for the unit test performance results.
+    :param endpoint: if specified, generate box plot for a specific test, otherwise, generate for all tests
+    :param form: the form that can be used for showing a subset of the data
+    :return:
+    """
     data = []
     with session_scope() as db_session:
-        suites = get_suites(db_session)
-    if not suites:
-        return None
-    for s in suites:
-        if test:
-            values = [str(c.execution_time) for c in get_test_measurements(db_session, name=test, suite=s.suite)]
+        if form:
+            suites = get_test_suites(db_session, limit=form.get_slider_value())
         else:
-            values = [str(c.execution_time) for c in get_measurements(db_session, suite=s.suite)]
-
-        data.append(go.Box(
-            x=values,
-            name="{0} ({1})".format(s.suite, len(values))))
-
-    layout = go.Layout(
-        autosize=True,
-        height=350 + 40 * len(suites),
-        plot_bgcolor='rgba(249,249,249,1)',
-        showlegend=False,
-        title='Execution times for every Travis build',
-        xaxis=dict(title='Execution time (ms)'),
-        yaxis=dict(
-            title='Build (measurements)',
-            autorange='reversed'
-        )
-    )
+            suites = get_test_suites(db_session)
 
-    return plotly.offline.plot(go.Figure(data=data, layout=layout), output_type='div', show_link=False)
+        if not suites:
+            return None
+        for s in suites:
+            if endpoint:
+                values = get_test_measurements(db_session, name=endpoint, suite=s.suite)
+            else:
+                values = get_suite_measurements(db_session, suite=s.suite)
+
+            data.append(go.Box(
+                x=values,
+                name='{} -'.format(s.suite)))
+
+        layout = go.Layout(
+            autosize=True,
+            height=350 + 40 * len(suites),
+            plot_bgcolor='rgba(249,249,249,1)',
+            showlegend=False,
+            xaxis=dict(title='Execution time (ms)'),
+            yaxis=dict(
+                title='Travis Build',
+                autorange='reversed'
+            )
+        )
+        return plotly.offline.plot(go.Figure(data=data, layout=layout), output_type='div', show_link=False)
```

Comment:
You can use 'core/plot/__init__.py' for this. In the view, don't import anything from plotly